### PR TITLE
fix When requesting /v1/text-to-audio with the streaming parameter set to true, get the TypeError: 'function' object is not iterable in tongyi tts model. #5251

### LINKED ISSUE #5251: When requesting /v1/text-to-audio with the streaming parameter set to true, get the TypeError: 'function' object is not iterable
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.0-fix1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Configure Tongyi Qianwen TTS model
image

2. Request /v1/text-to-audio url, set the parameter 'streaming' to true
curl --location 'http://localhost/v1/text-to-audio' \
--header 'accept: */*' \
--header 'Authorization: Bearer app-xxx' \
--data '{"text":"hello dify","streaming":true}'

### ✔️ Expected Behavior

Returns 200 with audio binary data

### ❌ Actual Behavior

Return code 500
image

### PR DESCRIPTION
**Pull request by yanyu510**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #5251 (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Fix my errors

**Review by ghost — APPROVED** [review]

File: api/core/model_runtime/model_providers/tongyi/tts/tts.py
Comment (yanyu510) [inline api/core/model_runtime/model_providers/tongyi/tts/tts.py:1]: already edited

**Comment by crazywoola** [comment]
Please fix the errors in CI

**Comment by crazywoola** [comment]
It seems this is not working with tongyi, I got a 500 error code when try to make a tts request. See logs below:

```
INFO:werkzeug:127.0.0.1 - - [19/Jun/2024 03:07:19] "OPTIONS /console/api/apps/2ee46612-d778-4219-a2f6-495f8caedcf5/text-to-audio HTTP/1.1" 200 -
ERROR:root:internal server error, 'NoneType' object has no attribute 'items'.
Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 119, in _tts_invoke
    if future.result():
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 220, in _process_sentence
    response = dashscope.audio.tts.SpeechSynthesizer.call(
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/audio/tts/speech_synthesizer.py", line 146, in call
    for part in response:
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/api_entities/websocket_request.py", line 71, in <genexpr>
    return (item for item in response)
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/common/utils.py", line 96, in async_to_sync
    for message in iter_over_async(async_generator, loop):
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/common/utils.py", line 86, in iter_over_async
    done, obj = loop.run_until_complete(get_next())
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/asyncio/base_events.py", line 625, in run_until_complete
    self._check_running()
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/asyncio/base_events.py", line 586, in _check_running
    raise RuntimeError(
RuntimeError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 122, in _tts_invoke
    raise InvokeBadRequestError(str(ex))
core.model_runtime.errors.invoke.InvokeBadRequestError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/tts_model.py", line 39, in invoke
    return self._invoke(model=model, credentials=credentials, user=user, streaming=streaming,
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 67, in _invoke
    return self._tts_invoke(model=model, credentials=credentials, content_text=content_text, voice=voice)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 136, in _tts_invoke
    raise InvokeBadRequestError(str(ex))
core.model_runtime.errors.invoke.InvokeBadRequestError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/app/audio.py", line 85, in post
    response = AudioService.transcript_tts(
  File "/Users/minibanana/Program/projects/dify/api/services/audio_service.py", line 111, in transcript_tts
    raise e
  File "/Users/minibanana/Program/projects/dify/api/services/audio_service.py", line 103, in transcript_tts
    return model_instance.invoke_tts(
  File "/Users/minibanana/Program/projects/dify/api/core/model_manager.py", line 283, in invoke_tts
    return self._round_robin_invoke(
  File "/Users/minibanana/Program/projects/dify/api/core/model_manager.py", line 303, in _round_robin_invoke
    return function(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/tts_model.py", line 42, in invoke
    raise self._transform_invoke_error(e)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/ai_model.py", line 69, in _transform_invoke_error
    for invoke_error, model_errors in self._invoke_error_mapping.items():
AttributeError: 'NoneType' object has no attribute 'items'
ERROR:app:Exception on /console/api/apps/2ee46612-d778-4219-a2f6-495f8caedcf5/text-to-audio [POST]
Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 119, in _tts_invoke
    if future.result():
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 220, in _process_sentence
    response = dashscope.audio.tts.SpeechSynthesizer.call(
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/audio/tts/speech_synthesizer.py", line 146, in call
    for part in response:
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/api_entities/websocket_request.py", line 71, in <genexpr>
    return (item for item in response)
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/common/utils.py", line 96, in async_to_sync
    for message in iter_over_async(async_generator, loop):
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/dashscope/common/utils.py", line 86, in iter_over_async
    done, obj = loop.run_until_complete(get_next())
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/asyncio/base_events.py", line 625, in run_until_complete
    self._check_running()
  File "/Users/minibanana/.pyenv/versions/3.10.14/lib/python3.10/asyncio/base_events.py", line 586, in _check_running
    raise RuntimeError(
RuntimeError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 122, in _tts_invoke
    raise InvokeBadRequestError(str(ex))
core.model_runtime.errors.invoke.InvokeBadRequestError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/tts_model.py", line 39, in invoke
    return self._invoke(model=model, credentials=credentials, user=user, streaming=streaming,
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 67, in _invoke
    return self._tts_invoke(model=model, credentials=credentials, content_text=content_text, voice=voice)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/tongyi/tts/tts.py", line 136, in _tts_invoke
    raise InvokeBadRequestError(str(ex))
core.model_runtime.errors.invoke.InvokeBadRequestError: Cannot run the event loop while another loop is running

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/app/audio.py", line 85, in post
    response = AudioService.transcript_tts(
  File "/Users/minibanana/Program/projects/dify/api/services/audio_service.py", line 111, in transcript_tts
    raise e
  File "/Users/minibanana/Program/projects/dify/api/services/audio_service.py", line 103, in transcript_tts
    return model_instance.invoke_tts(
  File "/Users/minibanana/Program/projects/dify/api/core/model_manager.py", line 283, in invoke_tts
    return self._round_robin_invoke(
  File "/Users/minibanana/Program/projects/dify/api/core/model_manager.py", line 303, in _round_robin_invoke
    return function(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/tts_model.py", line 42, in invoke
    raise self._transform_invoke_error(e)
  File "/Users/minibanana/Program/projects/dify/api/core/model_runtime/model_providers/__base/ai_model.py", line 69, in _transform_invoke_error
    for invoke_error, model_errors in self._invoke_error_mapping.items():
AttributeError: 'NoneType' object has no attribute 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/flask/views.py", line 110, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
  File "/Users/minibanana/Program/projects/dify/api/.venv/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/setup.py", line 86, in decorated
    return view(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/libs/login.py", line 91, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/wraps.py", line 21, in decorated
    return view(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/app/wraps.py", line 49, in decorated_view
    return view_func(*args, **kwargs)
  File "/Users/minibanana/Program/projects/dify/api/controllers/console/app/audio.py", line 116, in post
    raise InternalServerError()
werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
INFO:werkzeug:127.0.0.1 - - [19/Jun/2024 03:07:21] "POST /console/api/apps/2ee46612-d778-4219-a2f6-495f8caedcf5/text-to-audio HTTP/1.1" 500 -
```

**Review by crazywoola — CHANGES_REQUESTED** [review]
See comments

**Comment by yanyu510** [comment]
@crazywoola thx, I couldn't reproduce this issue in my local environment. can you tell me your operating environment?  
```python
 yanyu@yanyus-MacBook-Pro  ~/work/dify/dify_dev/dify/api   main ● ?  flask run --host 0.0.0.0 --port=5001 --debug 
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
load builtin provider identity=ToolProviderIdentity(author='Serply.io', name='websearch', description=I18nObject(zh_Hans='Serply.io 是一个强大的实时 SERP API，可提供来自 搜索 招聘 新闻等搜索引擎集合的结构化数据。', pt_BR='Serply.io is a robust real-time SERP API delivering structured data from a collection of search engines including Web Search, Jobs, News, and many more.', en_US='Serply.io is a robust real-time SERP API delivering structured data from a collection of search engines including Web Search, Jobs, News, and many more.'), icon='icon.svg', label=I18nObject(zh_Hans='Serply.io', pt_BR='Serply.io', en_US='Serply.io'), tags=[<ToolLabelEnum.SEARCH: 'search'>, <ToolLabelEnum.BUSINESS: 'business'>, <ToolLabelEnum.NEWS: 'news'>, <ToolLabelEnum.PRODUCTIVITY: 'productivity'>]) tools=None credentials_schema={'serply_api_key': ToolProviderCredentials(name='serply_api_key', type=<CredentialsType.SECRET_INPUT: 'secret-input'>, required=True, default=None, options=None, label=I18nObject(zh_Hans='Serply.io API key', pt_BR='Serply.io API key', en_US='Serply.io API key'), help=I18nObject(zh_Hans='从 Serply.io 获取您的 Serply.io API key', pt_BR='Get your Serply.io API key from Serply.io', en_US='Get your Serply.io API key from https://Serply.io/'), url='https://Serply.io/', placeholder=I18nObject(zh_Hans='请输入你的 Serply.io API key', pt_BR='Please input your Serply.io API key', en_US='Please input your Serply.io API key'))} error: 1 validation error for ScholarSearchTool
parameters.2.options.24.value
  Input should be a valid string [type=string_type, input_value=False, input_type=bool]
    For further information visit https://errors.pydantic.dev/2.7/v/string_type
2024-06-19 03:36:13.200 WARNING [MainThread] [ext_mail.py:50] - MAIL_TYPE is not set
2024-06-19 03:36:13.857 WARNING [MainThread] [_internal.py:97] -  * Debugger is active!
2024-06-19 03:36:13.876 INFO [MainThread] [_internal.py:97] -  * Debugger PIN: 685-857-456
2024-06-19 03:36:18.540 INFO [Thread-2 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:36:18] "POST /v1/text-to-audio HTTP/1.1" 200 -
2024-06-19 03:37:50.659 INFO [Thread-3 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:37:50] "OPTIONS /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio/voices?language=zh-Hans HTTP/1.1" 200 -
2024-06-19 03:37:50.697 INFO [Thread-4 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:37:50] "GET /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio/voices?language=zh-Hans HTTP/1.1" 200 -
2024-06-19 03:38:01.876 INFO [Thread-5 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:01] "POST /console/v1/text-to-audio HTTP/1.1" 404 -
2024-06-19 03:38:27.842 INFO [Thread-6 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:27] "OPTIONS /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio/voices?language=zh-Hans HTTP/1.1" 200 -
2024-06-19 03:38:27.866 INFO [Thread-7 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:27] "GET /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio/voices?language=zh-Hans HTTP/1.1" 200 -
2024-06-19 03:38:36.522 INFO [Thread-8 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:36] "OPTIONS /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/chat-messages HTTP/1.1" 200 -
2024-06-19 03:38:46.682 INFO [Thread-9 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:46] "POST /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/chat-messages HTTP/1.1" 200 -
2024-06-19 03:38:48.105 INFO [Thread-10 (_generate_worker)] [_client.py:1013] - HTTP Request: POST https://api.bianxieai.com/v1/chat/completions "HTTP/1.1 200 OK"
2024-06-19 03:38:48.172 INFO [Thread-11 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:48] "OPTIONS /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/chat-messages?conversation_id=6af60d01-763e-4581-bc1b-0139e43491ee HTTP/1.1" 200 -
2024-06-19 03:38:48.221 INFO [Thread-12 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:48] "GET /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/chat-messages?conversation_id=6af60d01-763e-4581-bc1b-0139e43491ee HTTP/1.1" 200 -
2024-06-19 03:38:54.212 INFO [Thread-13 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:54] "OPTIONS /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio HTTP/1.1" 200 -
2024-06-19 03:38:55.235 INFO [Thread-14 (process_request_thread)] [_internal.py:97] - 127.0.0.1 - - [19/Jun/2024 03:38:55] "POST /console/api/apps/10a0ea3b-d8fd-40df-9756-a5de9531f88e/text-to-audio HTTP/1.1" 200 -
```

**Comment by ic-xu** [comment]
The real-time performance of this method is relatively weak. It cannot generate text and play voice at the same time. We have reconstructed a version that can generate text and play voice at the same time. It will be sent to the community later. Then you can adjust it according to the new architecture. How about a modified version?

**Comment by yanyu510** [comment]
ok, understood. looking forward to it.

**Pull request closed without merging**